org.osgi.test.junit5/src/test/java/org/osgi/test/junit5/service/ServiceExtension_ServiceAwareTest.java

```diff
@@ -310,4 +310,56 @@ public void matchMultiple() throws Exception {
 
 		softly.assertAll();
 	}
+
+	static class ServiceClose extends TestBase {
+		@InjectService
+		ServiceAware<Foo> serviceAware;
+
+		@Override
+		Foo getService() {
+			return serviceAware.getService();
+		}
+
+		@Override
+		List<Foo> getServices() {
+			return serviceAware.getServices();
+		}
+
+		@Override
+		@Test
+		void test() throws Exception {
+			softly.assertThat(serviceAware.isEmpty())
+				.as("isEmpty %s", serviceAware)
+				.isFalse();
+			softly.assertThat(serviceAware.size())
+				.as("size %s", serviceAware)
+				.isEqualTo(1);
+			softly.assertThatCode(() -> serviceAware.close())
+				.doesNotThrowAnyException();
+			softly.assertThat(serviceAware.isEmpty())
+				.as("isEmpty %s", serviceAware)
+				.isTrue();
+			softly.assertThat(serviceAware.size())
+				.as("size %s", serviceAware)
+				.isEqualTo(0);
```

Comment:
When we see code like this it's often a hint that we should have an AssertJ class for the type under test. :smile:

I have pondered adding an AssertJ class for `ServiceAware`; if it would have utility beyond osgi-test's own internal test cases (I suspect it might) we should look at it.

Reply:
Sounds like an interesting addition for version 1.1 (after this release!).